File: tictactoechecker.py
def is_solved(board):
    pro = 0
    for row in board:
        for cell in row:
            if cell == 0:
                pro += 1

    one = 1
    two = 2

    a, b, c = board

    if (
        (a[0] == one and a[1] == one and a[2] == one) or
        (b[0] == one and b[1] == one and b[2] == one) or
        (c[0] == one and c[1] == one and c[2] == one) or
        (a[0] == one and b[0] == one and c[0] == one) or
        (a[1] == one and b[1] == one and c[1] == one) or
        (a[2] == one and b[2] == one and c[2] == one) or
        (a[0] == one and b[1] == one and c[2] == one) or
        (a[2] == one and b[1] == one and c[0] == one)
    ):
        return 1

    if (
        (a[0] == two and a[1] == two and a[2] == two) or
        (b[0] == two and b[1] == two and b[2] == two) or
        (c[0] == two and c[1] == two and c[2] == two) or
        (a[0] == two and b[0] == two and c[0] == two) or
        (a[1] == two and b[1] == two and c[1] == two) or
        (a[2] == two and b[2] == two and c[2] == two) or
        (a[0] == two and b[1] == two and c[2] == two) or
        (a[2] == two and b[1] == two and c[0] == two)
    ):
        return 2

    else:
        if pro > 0:
            return -1
        return 0

File: Javascript/TicTacToeChecker.js
function isSolved(board) {
    let pro = 0
    board.forEach(function(elt){
        elt.forEach(function(elts){
            if(elts === 0){
                pro++
            }
        })
    })
    
   
    
    let [a,b,c] = board
    let one = 1
    let two = 2
    
    if(
        (a[0] === one && a[1] === one && a[2] === one)||
        (b[0] === one && b[1] === one && b[2] === one)||
        (c[0] === one && c[1] === one && c[2] === one)||
        (a[0] === one && b[0] === one && c[0] === one)||
        (a[1] === one && b[1] === one && c[1] === one)||
        (a[2] === one && b[2] === one && c[2] === one)||
        (a[0] === one && b[1] === one && c[2] === one)||
        (a[2] === one && b[1] === one && c[0] === one)
        ){
            return 1
        }
        
    if(
        (a[0] === two && a[1] === two && a[2] === two)||
        (b[0] === two && b[1] === two && b[2] === two)||
        (c[0] === two && c[1] === two && c[2] === two)||
        (a[0] === two && b[0] === two && c[0] === two)||
        (a[1] === two && b[1] === two && c[1] === two)||
        (a[2] === two && b[2] === two && c[2] === two)||
        (a[0] === two && b[1] === two && c[2] === two)||
        (a[2] === two && b[1] === two && c[0] === two)
        ){
            return 2
        }
    

    else{
         if(pro > 0){
        return -1
    }
        return 0
    }
}

isSolved([[1,1,1],[2,1,2],[2,1,0]])